### DISCUSSION TITLE
Adding support for footnotes

### DISCUSSION THREAD

**Discussion by AlexRoidl** [aguingand/tiptap-markdown#37] [Q&A]
Hi there, 

thanks for this great plugin. I am struggling to get footnotes to work with the Markdown plugin. 
I looked into the extensions, but I am having a hard time to understand the full structure, to be able to implement it myself. 

Any help is appreciated!
Alex

**Comment by aguingand** [comment]
It's quite complicated because you must create 2 tiptap nodes (one inline for the reference + one the footnote) that handles the same DOM as https://github.com/markdown-it/markdown-it-footnote. Then you can check at [Custom extensions](https://github.com/aguingand/tiptap-markdown#custom-extensions) section to add markdown parse / serialize config.

**Reply by AlexRoidl** [reply]
yes, I see. Would the inline one be a node or a mark? and how do I prevent the markdown-it-footnote to interfere with each other, when I use it in two Nodes?

**Reply by AlexRoidl** [reply]
I managed to get a first draft running. Linking the tryout here, for anyone interested: https://github.com/AlexRoidl/tiptap-markdown-footnotes

**Reply by aguingand** [reply]
Nice !
